### ISSUE TITLE
Using reactDirective service renders components twice

### ISSUE THREAD

**Issue by hellsan631** [ngReact/ngReact#198]
I prefer using the reactDirective service because I get better control over my props, however when I use the reactDirective service, it causes each component to render twice, when values haven't changed.

This is not the behavior when using the react-component directive.

examples:

```js
var HelloComponent = React.createClass({
  propTypes: {
    fname : React.PropTypes.string.isRequired,
    lname : React.PropTypes.string.isRequired
  },
  render: function() {
    console.log('rendered');
    return <span>Hello {this.props.fname} {this.props.lname}</span>;
  }
})
app.value('HelloComponent', HelloComponent);
```
```html
 <react-component name="HelloComponent" props="person" watch-depth="reference"/>
```

Will output: ```rendered``` in the console, but

But when using the directive service:
```js
app.directive('helloComponent', function(reactDirective) {
  return reactDirective(HelloComponent);
});
```

```html
<hello-component fname="person.fname" lname="person.lname" />
```

Will output: 
```
rendered
rendered
```
in the console, meaning the component has rendered twice.

I think this is because the component is being rendered with the default props, and then being re-rendered on $scope watch initialization.

You should be able to get rid of line https://github.com/ngReact/ngReact/blob/master/ngReact.js#L290 while still getting the component to render, and it might fix this problem. Going to try to submit a PR for this later today.

**Comment by cjcenizal** [comment]
Yes! I just came here about this very issue. Thanks for logging this @hellsan631.

**Comment by marchq** [comment]
I am also noticing this issue. I happened to find it while doing some tracing in our React component. Is there any news on a fix?

There is also another issue. I'm noticing that when I don't use watch-depth, there is a render per prop even if it doesn't change.

I've attached a simple unit test that I based from DeepHello in tests/react-directive-spec.js. It logs the number of render calls without watch-depth. In this example it has three properties. It renders 4 times as seen in the console output:

```
LOG: 1
LOG: 2
LOG: 3
LOG: 4
```

[react-directive-rerender-spec.zip](https://github.com/ngReact/ngReact/files/2333582/react-directive-rerender-spec.zip)
